roscala/src/main/scala/coop/rchain/roscala/prim/Prim.scala

```diff
@@ -363,5 +371,45 @@ object Prim {
     */
   val map = Map(202 -> flPlus, 232 -> fxPlus)
 
+  def mismatchArgs(state: State, minArgs: Int, maxArgs: Int): Ob = {
+    val msg = if (maxArgs == MaxArgs) {
+      s"expected $minArgs or more arguments"
+    } else if (minArgs == maxArgs) {
+      if (minArgs == 1) {
+        "expected 1 argument"
+      } else {
+        s"expected $minArgs arguments"
+      }
+    } else {
+      s"expected between $minArgs and $maxArgs arguments"
+    }
+    return runtimeError(state.ctxt, msg)
```

Comment:
We defined  some case class representing error like below:

```scala
sealed trait PrimError
case class ArgumentMismatch(msg: String)               extends PrimError
case class TypeMismatch(argNum: Int, typeName: String) extends PrimError
case class IndexOutOfBounds(argNum: Int, size: Int)    extends PrimError
case object ArithmeticError                            extends PrimError
```
I choose to do the way as rosette did. I'm not clearly here, can you tell me which way is preferred.

Reply:
Yes, let's stay close to Rosette. If you discover already translated primitives that use `PrimError` let me know and we will decide what to do.